standards/HEALTH_CHECK_SPECIFICATION.md

```diff
@@ -0,0 +1,101 @@
+`CRITICAL` | Failure, your app will now be unable to function
+
+Health check endpoint
+---------------------
+
+The health check data is served via the health check endpoint.  This health check data comes in two forms.  The status code provides a quick way to get the overall status of the app, while the json body provides further information that might be useful in triaging any issues.
+
+The following requirements must be met by health check implementations:
+
+* All apps must have a health check endpoint
+* The health check endpoint must be available as a `GET /healthcheck` endpoint
+* The health check must return the [appropriate status code](#health-check-status-codes)
+* The health check must return [status information as json](#health-check-body)
+
+### Health check status codes
+
+There are three standard status codes that a health check can return:
+
+Status code | Description
+------------|------------------------
+`200`       | I'm fine
+`429`       | I'm warming up or I'm degraded
+`500`       | Kill me
+
+The health check should return 429 until the first checks have been run.  After that the codes returned should be based on the check status.  It is the responsibility of whatever is monitoring the health check to provide a grace period on startup.
+
+If any check returns a `CRITICAL` status (see [check statuses below](#check-statuses) then the app should transition to a `429`.  After a timeout that can be set per app the `429` will be downgraded `500`.  This delay in marking the app as `500` is to allow the issue a chance to resolve on its own before moving to a `500` which will result in the app being killed.
+
+If any check returns a `WARNING` status (see [check statuses below](#check-statuses) then the app should transition to a `429`.  In this situation the timeout does not apply and the app will continue to return a `429` until the status of the checks changes.
+
+Any status code ≥ 400 && != 429 will be treated as a failure.
+
+### Health check body
+
+The json body of the health check should implement the following spec:
+
+Field        | Type     | Description
+-------------|----------|---------------
+`status`     | `string` | The overall status of the health check using the values as the [check statuses](#check-statuses)
+`version`    | `string` | The version information of the app (e.g. commit, semver version, go version, build time, etc) as a string.
+`uptime`     | `ms`     | Milliseconds elapsed since the app started <sup>1</sup>
+`start_time` | `ISO8601`<sup>2</sup> | The time the app started in UTC
+`checks`     | [`[]Check`](#check)   | An array of the checks with details of their statuses
+
+<sup>1</sup> The start time of the app is actually approximate as we use the time at which the health check library was instantiated.
+
+<sup>2</sup> `ISO8601` UTC date time (`2006-01-02T15:04:05.999Z`)
+
+**Check**
+
+The following fields make up the details provided for each check in the array of checks within the json body.
```

Comment:
```suggestion
The following fields make up the details provided for each check-in the array of checks within the JSON body.
```

Reply:
`check in` should be left as is in this case as we are talking about each check that exists in the list rather than a `check-in`.  Casing changed on `JSON`.